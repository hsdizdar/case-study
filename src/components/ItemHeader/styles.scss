@import "../../styles/styles";

.container {
    display: flex;
    flex-direction: row;
    margin: 0px;
    padding: 0px 5px 0px 5px;
    background-color: transparent;
    border-bottom: 1px solid $border;

    &__left {
        display: flex;
        flex: 0.5;
        justify-content: space-between;
        align-items: center;
    }

    &__match-info {
        display: flex;
        flex-direction: row;
    }

    &__match-index {
        font-weight: bold;
        color: $index;
        margin-right: 5px;
    }

    &__match-date {
        font-weight: bold;
        color: $title;
        margin-right: 5px;
    }

    &__match-day {
        font-weight: bold;
        color: $title;
        margin-right: 5px;
    }

    &__match-league-name {
        font-weight: bold;
        color: $title;
        margin-right: 5px;
    }

    &__comment {
        flex: 1;
        text-align: end;
        border-right: 1px solid $border;
        padding: 10px 5px 10px 0px;
    }

    &__right {
        display: flex;
        flex: 1;
        justify-content: space-between;
        align-items: center;
    }

    &__item {
        display: flex;
        flex: 1;
        justify-content: space-between;
        align-items: center;
    }

    &__text {
        flex: 1;
        align-items: center;
        text-align: center;
        border-right: 1px solid $border;
        padding: 10px 0px 10px 0px;

        &--last-child {
            flex: 1;
            align-items: center;
            text-align: center;
            border-right: none;
        }

        &--empty {
            flex: 1;
            height: 36px;
            border-right: 1px solid $border;
            padding: 10px 0px 10px 0px;
        }
    }
}