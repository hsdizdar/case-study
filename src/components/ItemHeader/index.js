import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const ItemHeader = ({
  index, date, day, leagueName,
}) => (
  <div className="container">
    <div className="container__left">
      <div className="container__match-info">
        <p className="container__match-index">{index}</p>
        <p className="container__match-date">{date}</p>
        <p className="container__match-day">{day}</p>
        <p className="container__match-league-name">{leagueName}</p>
      </div>
      <p className="container__comment">Yorumlar</p>
    </div>
    <div className="container__right">
      <div className="container__item">
        <p className="container__text--empty" />
      </div>
      <div className="container__item">
        <p className="container__text">1</p>
        <p className="container__text">x</p>
        <p className="container__text">2</p>
      </div>
      <div className="container__item">
        <p className="container__text">Alt</p>
        <p className="container__text">Ãœst</p>
      </div>
      <div className="container__item">
        <p className="container__text">H1</p>
        <p className="container__text">1</p>
        <p className="container__text">x</p>
        <p className="container__text">2</p>
      </div>
      <div className="container__item">
        <p className="container__text">H2</p>
        <p className="container__text">1-X</p>
        <p className="container__text">X-2</p>
        <p className="container__text">2-X</p>
      </div>
      <div className="container__item">
        <p className="container__text">Var</p>
        <p className="container__text">Yok</p>
      </div>
      <div className="container__item">
        <p className="container__text--last-child">+99</p>
      </div>
    </div>
  </div>
);

ItemHeader.propTypes = {
  index: PropTypes.number,
  date: PropTypes.string,
  day: PropTypes.string,
  leagueName: PropTypes.string,
};

export default ItemHeader;
