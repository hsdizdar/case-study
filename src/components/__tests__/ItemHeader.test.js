import React from 'react';
import renderer from 'react-test-renderer';
import ItemHeader from '../ItemHeader';

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<ItemHeader />).toJSON();
    expect(tree).toMatchSnapshot();
});
