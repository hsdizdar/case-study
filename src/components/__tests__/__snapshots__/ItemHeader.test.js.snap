// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="container"
>
  <div
    className="container__left"
  >
    <div
      className="container__match-info"
    >
      <p
        className="container__match-index"
      />
      <p
        className="container__match-date"
      />
      <p
        className="container__match-day"
      />
      <p
        className="container__match-league-name"
      />
    </div>
    <p
      className="container__comment"
    >
      Yorumlar
    </p>
  </div>
  <div
    className="container__right"
  >
    <div
      className="container__item"
    >
      <p
        className="container__text--empty"
      />
    </div>
    <div
      className="container__item"
    >
      <p
        className="container__text"
      >
        1
      </p>
      <p
        className="container__text"
      >
        x
      </p>
      <p
        className="container__text"
      >
        2
      </p>
    </div>
    <div
      className="container__item"
    >
      <p
        className="container__text"
      >
        Alt
      </p>
      <p
        className="container__text"
      >
        Ãœst
      </p>
    </div>
    <div
      className="container__item"
    >
      <p
        className="container__text"
      >
        H1
      </p>
      <p
        className="container__text"
      >
        1
      </p>
      <p
        className="container__text"
      >
        x
      </p>
      <p
        className="container__text"
      >
        2
      </p>
    </div>
    <div
      className="container__item"
    >
      <p
        className="container__text"
      >
        H2
      </p>
      <p
        className="container__text"
      >
        1-X
      </p>
      <p
        className="container__text"
      >
        X-2
      </p>
      <p
        className="container__text"
      >
        2-X
      </p>
    </div>
    <div
      className="container__item"
    >
      <p
        className="container__text"
      >
        Var
      </p>
      <p
        className="container__text"
      >
        Yok
      </p>
    </div>
    <div
      className="container__item"
    >
      <p
        className="container__text--last-child"
      >
        +99
      </p>
    </div>
  </div>
</div>
`;
